src/brod_utils.erl

```diff
@@ -142,17 +142,23 @@ ensure_integer_offset_time(T) when is_integer(T) -> T.
 
 -spec do_find_leader_in_metadata(kpro_MetadataResponse(),
                                  topic(), partition()) -> endpoint().
-do_find_leader_in_metadata(Metadata, Topic, Partition) ->
+do_find_leader_in_metadata(Metadata, _Topic, Partition) ->
   #kpro_MetadataResponse{ broker_L        = Brokers
                         , topicMetadata_L = [TopicMetadata]
                         } = Metadata,
   #kpro_TopicMetadata{ errorCode           = TopicEC
                      , partitionMetadata_L = Partitions
                      } = TopicMetadata,
   kpro_ErrorCode:is_error(TopicEC) andalso erlang:throw(TopicEC),
-  #kpro_PartitionMetadata{leader = Id} =
-    lists:keyfind(Partition, #kpro_PartitionMetadata.partition, Partitions),
-  Id >= 0 orelse erlang:throw({no_leader, {Topic, Partition}}),
+  Id = case lists:keyfind(Partition,
+                          #kpro_PartitionMetadata.partition, Partitions) of
+         #kpro_PartitionMetadata{leader = Leader} when Leader >= 0 ->
+           Leader;
+         #kpro_PartitionMetadata{} ->
+           erlang:throw(?EC_LEADER_NOT_AVAILABLE);
+         _ ->
+           erlang:throw(?EC_UNKNOWN_TOPIC_OR_PARTITION)
```

Comment:
Then users will have more different kinds of errors to care about. I'd rather keep it simple.